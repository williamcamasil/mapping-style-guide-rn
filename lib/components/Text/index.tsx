import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
