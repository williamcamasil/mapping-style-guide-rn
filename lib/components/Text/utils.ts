import { TextStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType, FontWeightType, LineHeightType } from '../../tokens';
import { calcLineHeight } from '../../utils/typography';

export type TextVariantType =
  // heading
  'headingDisplay' |
  'headingXLarge' |
  'headingLarge' |
  'headingMedium' |
  'headingSmall' |
  // paragraph regular
  'subtitle' |
  'body' |
  'bodySmall' |
  'caption' |
  // small
  'small' |
  'smallCaps' |
  // one line
  'oneLineHuge' |
  'oneLineLarge' |
  'oneLineMedium' |
  'oneLineSmall' |
  'oneLineCaption' |
  'oneLineSemibold' |
  'oneLineBoldHuge' |
  'oneLineBoldExtraLarge' |
  'oneLineBoldLarge' |
  'oneLineBoldMedium' |
  'oneLineBoldSmall';

const getHeadingStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  headingDisplay: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.huge,
    lineHeight: calcLineHeight(theme.typography.sizes.huge, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingXLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXL, lineHeight ?? theme.typography.lineHeights.large),
  },
  headingMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.medium),
  },
});

const getSmallStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  small: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(14, lineHeight ?? theme.typography.lineHeights.medium),
  },
  smallCaps: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXS, lineHeight ?? theme.typography.lineHeights.medium),
    textTransform: 'uppercase' as TextStyle['textTransform'],
  },
});

const getOneLineStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  oneLineHuge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineLarge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineMedium: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineCaption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSemibold: {
    ...theme.typography.weights.semiBold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldHuge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldExtraLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
});

const getParagraphStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  subtitle: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.large),
  },
  bodySmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.large),
  },
  caption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.medium),
  },
  body: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.large),
  },
});

export function getVariantStyle(
  theme: AppThemeType,
  variant: TextVariantType,
  lineHeightName?: LineHeightType,
): TextStyle {
  const lineHeight = lineHeightName ? theme.typography.lineHeights[lineHeightName] : null;

  const variantStyles: Record<TextVariantType, TextStyle> = {
    // heading
    ...getHeadingStyles(theme, lineHeight),
    // small
    ...getSmallStyles(theme, lineHeight),
    // one line
    ...getOneLineStyles(theme, lineHeight),
    // paragraph
    ...getParagraphStyles(theme, lineHeight),
  };

  return variantStyles[variant];
}

export function getColorStyle(theme: AppThemeType, color: ColorPaletesType): TextStyle | null {
  return {
    color: theme.colors[color],
  };
}

export function getWeightStyle(theme: AppThemeType, weight?: FontWeightType): TextStyle | null {
  if (!weight) { return null; }

  return theme.typography.weights[weight];
}
