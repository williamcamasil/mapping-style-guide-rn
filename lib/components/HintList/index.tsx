import React from 'react';
import { StyleSheet, View } from 'react-native';

import { Images } from '../../assets';
import { useViewStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import { ColorPaletesType, SpacingSizesType } from '../../tokens';
import Spacer from '../Spacer';
import Text from '../Text';

export type HintListPropsType = {
  bulletPointColor: ColorPaletesType;
  bulletPointAlignCenter?: boolean;
  hintList: string[] | React.ReactNode[];
  theme: AppThemeType;
  isBold?: boolean;
  spacing?: SpacingSizesType;
};

const styles = StyleSheet.create({
  hintContainerStyle: {
    flexDirection: 'row',
  },
});

const HintList: React.FC<HintListPropsType> = ({
  bulletPointColor,
  bulletPointAlignCenter = true,
  hintList,
  theme,
  isBold = true,
  spacing = 'sXS',
}) => {

  const hintContainerStyles = useViewStyles(() => [
    styles.hintContainerStyle,
    {
      alignItems: bulletPointAlignCenter ? 'center' : 'flex-start',
    },
  ], [bulletPointAlignCenter]);

  const bulletPointStyles = useViewStyles(() => [
    {
      marginTop: bulletPointAlignCenter ? 0 : theme.spacings.sNano,
    },
  ], [bulletPointAlignCenter, theme.spacings.sNano]);

  const renderHint = (hint: string | React.ReactNode, index: number) => (
    <View key={`hint-${index}`}>
      <View style={hintContainerStyles}>
        <Images.BulletPoint color={bulletPointColor} style={bulletPointStyles} />
        <Spacer size={theme.spacings[spacing]} />
        <Text weight={isBold ? 'bold' : 'regular'}>{hint}</Text>
      </View>
      <Spacer size={theme.spacings.sSmall} />
    </View>
  );
  return (
    <View>
      {hintList.map(renderHint)}
    </View>
  );
};

export default withTheme(HintList);
