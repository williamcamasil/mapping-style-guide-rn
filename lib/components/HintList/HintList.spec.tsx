import React from 'react';

import { render } from '@testing-library/react-native';

import HintList from '.';
import Text from '../Text';

describe('HintList', () => {
  it('Should match default HintList snapshot', () => {
    const result = render(
      <HintList
        bulletPointColor="neutralGray400"
        hintList={[
          'hint 1',
          'hint 2',
        ]}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('Should match custom HintList snapshot', () => {
    const result = render(
      <HintList
        bulletPointColor="neutralGray400"
        hintList={[
          'hint 1',
          'hint 2',
        ]}
        isBold={false}
        spacing="sXXS"
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('Should match custom HintList snapshot with prop bulletPointAlignCenter', () => {
    const result = render(
      <HintList
        bulletPointColor="neutralGray400"
        hintList={[
          'hint 1',
          'hint 2',
          (<Text weight="bold">Hint 3</Text>),
        ]}
        isBold={false}
        spacing="sXXS"
        bulletPointAlignCenter={false}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

});
