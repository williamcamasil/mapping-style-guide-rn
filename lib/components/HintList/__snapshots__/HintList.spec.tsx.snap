// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintList Should match custom HintList snapshot 1`] = `
<View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 0,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 1
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 0,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 2
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`HintList Should match custom HintList snapshot with prop bulletPointAlignCenter 1`] = `
<View>
  <View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 8,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 1
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 8,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 2
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 8,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 12,
            "width": 12,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "bold",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Hint 3
        </Text>
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`HintList Should match default HintList snapshot 1`] = `
<View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 0,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 16,
            "width": 16,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 1
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="8"
        bbWidth="8"
        fill="none"
        focusable={false}
        height={8}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "marginTop": 0,
            },
            {
              "flex": 0,
              "height": 8,
              "width": 8,
            },
          ]
        }
        vbHeight={8}
        vbWidth={8}
        width={8}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="4"
            cy="4"
            fill={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
            r="4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 16,
            "width": 16,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        hint 2
      </Text>
    </View>
    <Text
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
